Fri Mar 26 15:26:25 EET 2010  Sergei Trofimovich <slyfox@community.haskell.org>
  * Makefile.am: added missing '_stub.o' files
  
  Failing to do so leads to nonworking gtk2hs packages:
  $ ghci -package gio
  ghc: HSgio.o: unknown symbol `SystemziGIOziFile_d30J'
diff -rN -u old-gtk2hs/configure.ac new-gtk2hs/configure.ac
--- old-gtk2hs/configure.ac	2010-03-26 15:30:57.041527760 +0200
+++ new-gtk2hs/configure.ac	2010-03-26 15:30:58.714528600 +0200
@@ -478,6 +478,7 @@
 HAVE_GTK_VERSION_2_8=`$PKG_CONFIG gtk+-2.0 --atleast-version=2.8 && echo yes || echo no`
 HAVE_GTK_VERSION_2_10=`$PKG_CONFIG gtk+-2.0 --atleast-version=2.10 && echo yes || echo no`
 HAVE_GTK_VERSION_2_12=`$PKG_CONFIG gtk+-2.0 --atleast-version=2.12 && echo yes || echo no`
+HAVE_GTK_VERSION_2_14=`$PKG_CONFIG gtk+-2.0 --atleast-version=2.14 && echo yes || echo no`
 
 AM_CONDITIONAL(HAVE_GTK_VERSION_2_2, test "$HAVE_GTK_VERSION_2_2" = "yes")
 AM_CONDITIONAL(HAVE_GTK_VERSION_2_4, test "$HAVE_GTK_VERSION_2_4" = "yes")
@@ -485,6 +486,7 @@
 AM_CONDITIONAL(HAVE_GTK_VERSION_2_8, test "$HAVE_GTK_VERSION_2_8" = "yes")
 AM_CONDITIONAL(HAVE_GTK_VERSION_2_10, test "$HAVE_GTK_VERSION_2_10" = "yes")
 AM_CONDITIONAL(HAVE_GTK_VERSION_2_12, test "$HAVE_GTK_VERSION_2_12" = "yes")
+AM_CONDITIONAL(HAVE_GTK_VERSION_2_14, test "$HAVE_GTK_VERSION_2_14" = "yes")
 
 CREATE_TYPES="$CREATE_TYPES `test "$HAVE_GTK_VERSION_2_2" = "yes" && echo gtk-2.2`"
 CREATE_TYPES="$CREATE_TYPES `test "$HAVE_GTK_VERSION_2_4" = "yes" && echo gtk-2.4`"
@@ -492,6 +494,7 @@
 CREATE_TYPES="$CREATE_TYPES `test "$HAVE_GTK_VERSION_2_8" = "yes" && echo gtk-2.8`"
 CREATE_TYPES="$CREATE_TYPES `test "$HAVE_GTK_VERSION_2_10" = "yes" && echo gtk-2.10`"
 CREATE_TYPES="$CREATE_TYPES `test "$HAVE_GTK_VERSION_2_12" = "yes" && echo gtk-2.12`"
+CREATE_TYPES="$CREATE_TYPES `test "$HAVE_GTK_VERSION_2_14" = "yes" && echo gtk-2.14`"
 
 dnl Also allow us to conditionally compile binding to the new Gtk+ APIs.
 GTKHS_PACKAGE_ADD_CHECK_VERSION(gtk+-2.0, GTK, Gtk+)
diff -rN -u old-gtk2hs/Makefile.am new-gtk2hs/Makefile.am
--- old-gtk2hs/Makefile.am	2010-03-26 15:30:56.983528600 +0200
+++ new-gtk2hs/Makefile.am	2010-03-26 15:30:58.695528560 +0200
@@ -454,7 +454,8 @@
 	$(libHSgio_a_GENERATEDSOURCES:.chs=.hs)
 
 libHSgio_a_LIBADD  = 				\
-	gio/System/GIO/Base_stub.o
+	gio/System/GIO/Base_stub.o		\
+	gio/System/GIO/File_stub.o
 
 if !USE_GCLOSUE_SIGNALS_IMPL
 libHSgio_a_LIBADD +=	\
@@ -885,6 +886,19 @@
 	gtk/Graphics/UI/Gtk/Windows/AboutDialog_stub.o
 endif
 
+if HAVE_GTK_VERSION_2_10
+libHSgtk_a_LIBADD +=							\
+	gtk/Graphics/UI/Gtk/Buttons/LinkButton_stub.o			\
+	gtk/Graphics/UI/Gtk/Layout/Notebook_stub.o			\
+	gtk/Graphics/UI/Gtk/Multiline/TextBuffer_stub.o			\
+	gtk/Graphics/UI/Gtk/Windows/Assistant_stub.o
+endif
+
+if HAVE_GTK_VERSION_2_14
+libHSgtk_a_LIBADD +=							\
+	gtk/Graphics/UI/Gtk/Misc/Calendar_stub.o
+endif
+
 if !USE_GCLOSUE_SIGNALS_IMPL
 libHSgtk_a_LIBADD += gtk/Graphics/UI/Gtk/Signals_stub.o
 endif
